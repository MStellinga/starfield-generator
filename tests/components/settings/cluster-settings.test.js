/* eslint-disable jest/valid-expect */
import { h } from 'preact';
import ClusterSettings from '../../../src/components/settings/cluster-settings';
import render from 'preact-render-to-string';

describe('Test Cluster settings', () => {
	test('Cluster settings renders', () => {
        const tree = render(<ClusterSettings clusters={[]} />);
        expect(tree).toMatchSnapshot();	
    });

    test('Cluster settings with some clusters renders', () => {
        const tree = render(<ClusterSettings clusters={[
            {x: 1, y: 5, strength: 300, r: 0, g: 0, b:250,a:255, size1stars:42, size2stars:50, size3stars:10, bubbles:500, generated:0, fractalSize: 0 },
            {x: 3, y: 28, strength: 300, r: 180, g: 0, b:180,a:255, size1stars:0, size2stars:0, size3stars:0, bubbles:0, generated:0, fractalSize: 100 }
        ]} />);
        expect(tree).toMatchSnapshot();	
    });    

    // Unfortunately, adding tests that call event handlers is hard to do with Enzyme and function-based Preact components (at this time, anyway)
});
