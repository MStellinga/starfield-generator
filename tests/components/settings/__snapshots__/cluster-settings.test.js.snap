// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cluster settings Cluster settings renders 1`] = `"<div class=\\"section\\"><h2>Clusters</h2><table class=\\"clusters\\"><thead><tr><th></th><th>Power</th><th>Stars 1</th><th>Stars 2</th><th>Stars 3</th><th>Bubbles</th><th>Fractal</th><th>Color</th><th></th></tr></thead><tbody></tbody></table></div>"`;

exports[`Test Cluster settings Cluster settings with some clusters renders 1`] = `"<div class=\\"section\\"><h2>Clusters</h2><table class=\\"clusters\\"><thead><tr><th></th><th>Power</th><th>Stars 1</th><th>Stars 2</th><th>Stars 3</th><th>Bubbles</th><th>Fractal</th><th>Color</th><th></th></tr></thead><tbody><tr><td></td><td></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"42\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"50\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"10\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"500\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><div><div class=\\"pickerButton\\" style=\\"background-color: rgb(0,0,250);\\"></div></div></td><td></td></tr><tr><td>1</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"300\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"100\\" /></td><td><div><div class=\\"pickerButton\\" style=\\"background-color: rgb(180,0,180);\\"></div></div></td><td><button>X</button></td></tr></tbody></table></div>"`;
